homeassistant/components/zwave/util.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def value_handler(value, method=None, class_id=None, index=None,
```

Comment:
So just a few things about this method.

I moved it to a util class because it only used `self` for the value. Passing in the value allowed us to extract it. Will make it a lot easier to test 👍 

However, this method should be broken into two parts: one that gets values and one that sets it. Right now it makes this method way too complicated, too many things crammed into a method.

I didn't want to do it because I am unable to test it locally. Instead, leaving it here hoping someone picks it up 👍  

```python
def _set_value(value, data, **kwargs):
    value = _get_value(value, **kwargs)
    # Some error checking
    value.data = data
```